src/Adapter/Adapter.php

```diff
@@ -177,18 +177,17 @@ public function query(
         }
 
         if ($mode === self::QUERY_MODE_PREPARE) {
-            $this->lastPreparedStatement = null;
-            $this->lastPreparedStatement = $this->driver->createStatement($sql);
-            $this->lastPreparedStatement->prepare();
+            $lastPreparedStatement = $this->driver->createStatement($sql);
+            $lastPreparedStatement->prepare();
             if (is_array($parameters) || $parameters instanceof ParameterContainer) {
                 if (is_array($parameters)) {
-                    $this->lastPreparedStatement->setParameterContainer(new ParameterContainer($parameters));
+                    $lastPreparedStatement->setParameterContainer(new ParameterContainer($parameters));
                 } else {
-                    $this->lastPreparedStatement->setParameterContainer($parameters);
+                    $lastPreparedStatement->setParameterContainer($parameters);
                 }
-                $result = $this->lastPreparedStatement->execute();
+                $result = $lastPreparedStatement->execute();
             } else {
-                return $this->lastPreparedStatement;
+                return $lastPreparedStatement;
```

Comment:
Overall OK with removing stateful changes to `$this`, but this changes the behavior on the adapter itself, no? is `$this->lastPreparedStatement` used elsewhere, or was it only here for debugging purposes?

Reply:
`$this->lastPreparedStatement` was used only in this place. From my understanding should not change the behavior of the adapter as it was used only in this code block.